doc/tutorials/your_first_project/your_first_project.rst

```diff
@@ -0,0 +1,276 @@
+
+This first bit is a bit of boilerplate but you should be used to seeing this from the ROS 2 tutorials.
+
+.. code-block:: C++
+
+  #include <memory>
+  #include <thread>
+  #include <string>
+
+  #include <rclcpp/rclcpp.hpp>
+  #include <moveit/move_group_interface/move_group_interface.h>
+
+  int main(int argc, char * argv[])
+  {
+    // Initialize ROS and create the Node
+    rclcpp::init(argc, argv);
+    const auto node = std::make_shared<rclcpp::Node>(
+      "hello_moveit",
+      rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+    );
+
+    // Create a ROS logger
+    const auto logger = rclcpp::get_logger("hello_moveit");
+
+    // Create a thread to run to spin a Executor
+    auto thread = std::thread([node]() {
+      rclcpp::executors::SingleThreadedExecutor executor;
+      executor.add_node(node);
+      executor.spin();
+    });
+
+    // Next step goes here
+
+    // Shutdown ROS and our thread
+    rclcpp::shutdown();
+    thread.join();
+    return 0;
+  }
+
+2.1 Build and Run
+~~~~~~~~~~~~~~~~~
+
+We will build and run the program to see that everything is right before we move on.
+
+Change directory back to the workspace directory ``ws_moveit2`` and run this command:
+
+.. code-block:: bash
+
+  colcon build --mixin debug
+
+After this succeeds, **open a new terminal**, then source the workspace environment script in that new terminal so that we can run our program.
+
+.. code-block:: bash
+
+  source install/setup.bash
+
+Run your program and see the output.
+
+.. code-block:: bash
+
+  ros2 run hello_moveit hello_moveit
+
+You will probably see this error:
+
+.. code-block::
+
+  terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
+    what():  the given context is not valid, either rcl_init() was not called or rcl_shutdown() was called., at /tmp/binarydeb/ros-rolling-rcl-4.0.0/src/rcl/guard_condition.c:67
+  [ros2run]: Aborted
+
+This is because ``rclcpp::shutdown()`` was called before the Executor finished initializing.
+We'll fix that in the next section by adding some code that does something after creating the thread.
+
+2.2 Examine the code
+~~~~~~~~~~~~~~~~~~~~
+
+The headers included at the top are just some standard C++ headers and the header for ROS and MoveIt that we will use later.
+
+After that we have the normal call to initialize rclcpp and then we create our Node.
+
+.. code-block:: C++
+
+  const auto node = std::make_shared<rclcpp::Node>(
+    "hello_moveit",
+    rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+  );
+
+The first argument is the string that ROS will use to make a unique node.
+The second is needed for MoveIt because of how we use ROS Parameters.
+
+Before we can start use our ROS Node we have to give it to a spinning executor.
+This is what will enable ROS to call callbacks to update our Robot State among other things.
+
+.. code-block:: C++
+
+  auto thread = std::thread([=]() {
```

Comment:
Sorry for the drive by, just looking around for stuff and saw this, but earlier in the code snippet it says `auto thread = std::thread([node]() {`. and it says `=`. Not a big deal but I thought I'd point it out...

Reply:
That is a good point.  I should audit this and make sure it is consistent.  Right now we are working on fixing a race condition in MoveGroupInterface.  Once that is fixed I'll clean this up and add it as a tutorial.

Reply:
I just updated this tutorial to fix this.  All of the explanations and the larger blocks should be consistent.  I would really appreciate a review of that if you have time.